components/base/sensorcontrolled/sensorcontrolled.go

```diff
@@ -158,11 +171,47 @@ func (sb *sensorBase) Reconfigure(ctx context.Context, deps resource.Dependencie
 		return errors.Wrapf(err, "no base named (%s)", newConf.Base)
 	}
 
-	if sb.velocities != nil && useControlLoop {
-		sb.controlLoopConfig = sb.createControlLoopConfig()
-		if err := sb.setupControlLoops(); err != nil {
-			return err
+	if sb.velocities != nil && len(newConf.ControlParameters) != 0 {
+		// assign linear and angular PID correctly based on the given type
+		var linear, angular basePIDConfig
+		for _, c := range newConf.ControlParameters {
+			if c.Type == typeLinVel {
+				linear = c
+			} else if c.Type == typeAngVel {
```

Comment:
Since we are only comparing between typeLinVel and typeAngVel, I would recommend that you get rid of the else if check and assign c to angular in an else statement instead. 
```
if c.Type == typeLinVel {
	linear = c
} else {
       angular = c 
}
```

Reply:
hm I think instead I might add a warning that the types aren't correct in an `else` block. since these are coming from the config I don't just want to assume anything not linear is angular